pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,475 @@
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+	}
+	log.Info("Reconcile of deadmanssnitch integration complete")
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileDeadmansSnitchIntegration) getMatchingClusterDeployment(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration) (*hivev1.ClusterDeploymentList, error) {
+	selector, err := metav1.LabelSelectorAsSelector(&dmsi.Spec.ClusterDeploymentSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	matchingClusterDeployments := &hivev1.ClusterDeploymentList{}
+	listOpts := &client.ListOptions{LabelSelector: selector}
+	err = r.client.List(context.TODO(), matchingClusterDeployments, listOpts)
+	return matchingClusterDeployments, err
+}
+
+// Add finalizers to both the deadmanssnitch integreation and the matching cluster deployment
+func (r *ReconcileDeadmansSnitchIntegration) dmsAddFinalizer(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clusterdeployment *hivev1.ClusterDeployment) error {
+	deadMansSnitchFinalizer := "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clusterdeployment.Name, "cluster-deployment.Namespace:", clusterdeployment.Namespace)
+	//checking i finalizers exits in the clusterdeployment adding if they dont
+	logger.Info("Checking for finalizers")
+	if utils.HasFinalizer(clusterdeployment, deadMansSnitchFinalizer) == false {
+		log.Info(fmt.Sprint("Adding finalizer to cluster Deployment Name:  ", clusterdeployment.Name+" namespace:"+clusterdeployment.Namespace+" DMSI Name  :"+dmsi.Name))
+		utils.AddFinalizer(clusterdeployment, deadMansSnitchFinalizer)
+		err := r.client.Update(context.TODO(), clusterdeployment)
+		if err != nil {
+			return err
+		}
+
+	}
+	logger.Info("Cluster deployment finalizer created nothing to do here ...: ")
+
+	//checking i finalizers exits in the dmsi cr adding if they dont
+	logger.Info("Checking for finalizers")
+	if utils.HasFinalizer(dmsi, deadMansSnitchFinalizer) == false {
+		log.Info(fmt.Sprint("Adding finalizer to DMSI Name: ", " DMSI Name: :"+dmsi.Name))
+		utils.AddFinalizer(dmsi, deadMansSnitchFinalizer)
+		err := r.client.Update(context.TODO(), dmsi)
+		if err != nil {
+			return err
+		}
+
+	}
+	logger.Info("DMSI finalizer created nothing to do here..: ")
+
+	return nil
+
+}
+
+// create snitch in deadmanssnitch.com with information retrived from dmsi cr as well as the matching cluster deployment
+func (r *ReconcileDeadmansSnitchIntegration) createSnitch(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, cd *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+	snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
```

Comment:
```suggestion
	snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain
	if dmsi.Spec.SnitchNamePostFix != "" {
		snitchName += + "-" + dmsi.Spec.SnitchNamePostFix
	}
```

Need to support no postfix for snitches, it's an optional argument.  We don't have any postfix for osd today.

Reply:
I know the postfix was made optional a few commits back but im thinking now it might cause problems down the road with DMS name duplication.

Reply:
@jewzaam

Reply:
Ok cool ill make the suggested changes